### PR TITLE
Remove Cancel on card reader updates at 100%

### LINKED ISSUE #5324: [Mobile Payments] [SDK 2] Software Updated, 100% Complete, Should Probably Not Show Cancel Button
See also pdi1mn-2k-p2

Found during testing of https://github.com/woocommerce/woocommerce-ios/pull/5322

When a reader software update completes, it sits on this screen for 10-15 seconds. I think we should consider NOT showing the Cancel button once the update is 100% complete, to avoid the merchant being tempted to intervene when the update is so close to being "done done."

IMG_0223

### PR DESCRIPTION
**Pull request by joshheald**
Closes #5324 

## Description

Card reader updates hover at 100% when they are almost complete, displaying 100% in the progress view, for up to 15s. We've taken the decision to remove the Cancel button once 100% is displayed, to avoid a merchant ending an update right before it finishes.

## Testing
1. Simulate an optional update using the instructions here: PdfdoF-fr-p2
2. Go to `⚙️ > In-Person Payments > Manage Card Reader`
3. Connect to the card reader simulator
4. Start the update
5. Observe that the cancel button is removed as soon as the progress indicator gets to 100%
6. Redo the update, and confirm that before 100% the cancel button functions as expected.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift [hunk 1]
```diff
@@ -127,9 +127,10 @@ private extension CardReaderSettingsConnectedViewController {
             }
         } else if let readerUpdateProgress = viewModel.readerUpdateProgress {
             // If we are updating a reader, show the progress alert
-            settingsAlerts.updateProgress(from: self, requiredUpdate: false, progress: readerUpdateProgress, cancel: { [weak self] in
-                self?.viewModel?.cancelCardReaderUpdate()
-            })
+            settingsAlerts.updateProgress(from: self,
+                                          requiredUpdate: false,
+                                          progress: readerUpdateProgress,
+                                          cancel: viewModel.cancelCardReaderUpdate)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift:133]: nit: it took me some time to realize why this change hides the cancel CTA (by whether `cancel` closure is nil or not), would it be more clear if there is another boolean from the view model like `isCancelButtonShown`/`showsCancelButton` based on the progress? This way, the cancel closure can stay the same and we just change the cancel button visibility boolean.

**Review by jaclync — APPROVED** [review]
LGTM! Code and testing ✅ (just a suggestion for your consideration)

Some non-blocking side notes:
- During testing, I have a habit of checking memory debugger on code changes with a closure that references self. I noticed that `CardReaderSettingsConnectedViewController` was still in the memory after leaving the payments screen, and saw the retain cycle at a place unrelated to this PR:
https://github.com/woocommerce/woocommerce-ios/blob/dbea5342d2b1604904d109d6feb860b799b685d1/WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift#L191-L193
Passing weak self there deallocates the view controller after leaving the screen.
- Minor visual suggestion: there seems to be quite some space at the bottom when the cancel CTA is not shown, though this is for like a second so it might not be worth the effort.

**Comment by joshheald** [comment]
> * During testing, I have a habit of checking memory debugger on code changes with a closure that references self. I noticed that `CardReaderSettingsConnectedViewController` was still in the memory after leaving the payments screen, and saw the retain cycle at a place unrelated to this PR:
>   https://github.com/woocommerce/woocommerce-ios/blob/dbea5342d2b1604904d109d6feb860b799b685d1/WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift#L191-L193
>   
>   Passing weak self there deallocates the view controller after leaving the screen.

My instant suspicion is that you've just found the cause of #5332, which is next on my list! 🙇‍♂️ 


> * Minor visual suggestion: there seems to be quite some space at the bottom when the cancel CTA is not shown, though this is for like a second so it might not be worth the effort.

Yeah, I know what you mean, but I think it's best to leave it as is in this case since it's so short a time.
